mne/viz/_figure.py

```diff
@@ -0,0 +1,1854 @@
+                sensor_picks.extend(ch_indices[this_type])
+        sensor_idx = np.in1d(sensor_picks, pick).nonzero()[0]
+        # change the sensor color
+        fig = self.mne.fig_selection
+        fig.lasso.ec[sensor_idx, 0] = float(mark_bad)  # change R of RGBA array
+        fig.lasso.collection.set_edgecolors(fig.lasso.ec)
+        fig.canvas.draw_idle()
+
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+    # PROJECTORS & BAD CHANNELS
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+    def _create_proj_fig(self):
+        """Create the projectors dialog window."""
+        from matplotlib.widgets import Button, CheckButtons
+
+        projs = self.mne.projs
+        labels = [p['desc'] for p in projs]
+        for ix, active in enumerate(self.mne.projs_active):
+            if active:
+                labels[ix] += ' (already applied)'
+        # make figure
+        width = max([4.5, max([len(label) for label in labels]) / 8 + 0.5])
+        height = (len(projs) + 1) / 6 + 1.5
+        fig = self._new_child_figure(figsize=(width, height),
+                                     fig_name='fig_proj',
+                                     window_title='SSP projection vectors')
+        # make axes
+        offset = (1 / 6 / height)
+        position = (0, offset, 1, 0.8 - offset)
+        ax = fig.add_axes(position, frame_on=False, aspect='equal')
+        # make title
+        first_line = ('Projectors already applied to the data are dimmed.\n'
+                      if any(self.mne.projs_active) else '')
+        second_line = 'Projectors marked with "X" are active on the plot.'
+        ax.set_title(f'{first_line}{second_line}')
+        # draw checkboxes
+        checkboxes = CheckButtons(ax, labels=labels, actives=self.mne.projs_on)
+        # gray-out already applied projectors
+        for label, rect, lines in zip(checkboxes.labels,
+                                      checkboxes.rectangles,
+                                      checkboxes.lines):
+            if label.get_text().endswith('(already applied)'):
+                label.set_color('0.5')
+                rect.set_edgecolor('0.7')
+                [x.set_color('0.7') for x in lines]
+            rect.set_linewidth(1)
+        # add "toggle all" button
+        ax_all = fig.add_axes((0.25, 0.01, 0.5, offset), frame_on=True)
+        fig.mne.proj_all = Button(ax_all, 'Toggle all')
+        # add event listeners
+        checkboxes.on_clicked(self._toggle_proj_checkbox)
+        fig.mne.proj_all.on_clicked(partial(self._toggle_proj_checkbox,
+                                            toggle_all=True))
+        # save params
+        fig.mne.proj_checkboxes = checkboxes
+        # show figure (this should work for non-test cases)
+        try:
+            self.mne.fig_proj.canvas.draw()
+            plt_show(fig=self.mne.fig_proj, warn=False)
+        except Exception:
+            pass
```

Comment:
Do we still need this try/except? I think I recently pushed some fixes to our `plt_show` that might make it so you don't...?

Reply:
not sure, let's find out :)